CustomControlLibrary/exchange/IExchangeRateProvider.cs

```diff
@@ -1,12 +1,13 @@
-﻿using System.Threading;
+﻿using System;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace CustomControlLibrary
 {
     /// <summary>
     /// Provides a exchange currency rate
     /// </summary>
-    public interface IExchangeRateProvider
+    public interface IExchangeRateProvider: IDisposable
```

Comment:
please note that in WPF IDisposble is called automatically in very rare situations, in most cases you should call it manually when unloading control or something like this. It means that you should take care about initializing provider again if control is loaded/unloaded many times

Reply:
Yes. Suspect that parent of control will create and destroy instance of IExchangeRateProvider. This demo-application less logic which requires destroying of provider.